File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityQuestionsActivity.java
package com.eulersbridge.isegoria.personality;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.eulersbridge.isegoria.R;
import com.eulersbridge.isegoria.util.ui.NonSwipeableViewPager;
import com.eulersbridge.isegoria.util.ui.SimpleFragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class PersonalityQuestionsActivity extends AppCompatActivity {

    private NonSwipeableViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.personality_questions_activity);

        PersonalityViewModel viewModel = ViewModelProviders.of(this).get(PersonalityViewModel.class);

        viewModel.userSkippedQuestions.observe(this, skipped -> {
            if (skipped != null && skipped)
                finish();
        });

        viewModel.userContinuedQuestions.observe(this, continued -> {
            if (continued != null && continued)
                viewPager.setCurrentItem(1);
        });

        viewModel.userCompletedQuestions.observe(this, completed -> {
            if (completed != null && completed)
                finish();
        });

        setupViewPager();
    }

    private void setupViewPager() {
        viewPager = findViewById(R.id.personality_view_pager);

        List<Fragment> fragments = new ArrayList<>();
        fragments.add(new PersonalityPermissionFragment());
        fragments.add(new PersonalityQuestionsFragment());

        SimpleFragmentPagerAdapter viewPagerAdapter = new SimpleFragmentPagerAdapter(getSupportFragmentManager(), fragments);
        viewPager.setAdapter(viewPagerAdapter);

        viewPager.setCurrentItem(0);
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityQuestionsActivity.kt
package com.eulersbridge.isegoria.personality

import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.eulersbridge.isegoria.R
import com.eulersbridge.isegoria.observe
import com.eulersbridge.isegoria.util.ui.NonSwipeableViewPager
import com.eulersbridge.isegoria.util.ui.SimpleFragmentPagerAdapter
import kotlinx.android.synthetic.main.personality_questions_activity.*

class PersonalityQuestionsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.personality_questions_activity)

        val viewModel = ViewModelProviders.of(this).get(PersonalityViewModel::class.java)

        observe(viewModel.userSkippedQuestions) {
            if (it == true) finish()
        }

        observe(viewModel.userContinuedQuestions) {
            if (it == true) getViewPager().currentItem = 1
        }

        observe(viewModel.userCompletedQuestions) {
            if (it == true) finish()
        }

        setupViewPager()
    }

    private fun getViewPager() = viewPager as NonSwipeableViewPager

    private fun setupViewPager() {
        val fragments = listOf(
            PersonalityPermissionFragment(),
            PersonalityQuestionsFragment()
        )

        val viewPagerAdapter = SimpleFragmentPagerAdapter(supportFragmentManager, fragments)

        getViewPager().apply {
            adapter = viewPagerAdapter
            currentItem = 0
        }
    }
}